process/firehoseDataProcessor.go

```diff
@@ -0,0 +1,137 @@
+package process
+
+import (
+	"encoding/hex"
+	"fmt"
+	"io"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+	"github.com/multiversx/mx-chain-core-go/data/outport"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	logger "github.com/multiversx/mx-chain-logger-go"
+)
+
+var log = logger.GetOrCreate("firehose")
+
+const (
+	firehosePrefix   = "FIRE"
+	beginBlockPrefix = "BLOCK_BEGIN"
+	endBlockPrefix   = "BLOCK_END"
+)
+
+type dataProcessor struct {
+	marshaller        marshal.Marshalizer
+	operationHandlers map[string]func(marshalledData []byte) error
+	writer            io.Writer
+	blockCreator      BlockContainerHandler
+}
+
+// NewFirehoseDataProcessor creates a data processor able to receive data from a ws outport driver and print saved blocks
+func NewFirehoseDataProcessor(
+	writer io.Writer,
+	blockCreator BlockContainerHandler,
+	marshaller marshal.Marshalizer,
+) (DataProcessor, error) {
+	if writer == nil {
+		return nil, errNilWriter
+	}
+	if check.IfNil(blockCreator) {
+		return nil, errNilBlockCreator
+	}
+	if check.IfNil(marshaller) {
+		return nil, errNilMarshaller
+	}
+
+	dp := &dataProcessor{
+		marshaller:   marshaller,
+		writer:       writer,
+		blockCreator: blockCreator,
+	}
+
+	dp.operationHandlers = map[string]func(marshalledData []byte) error{
+		outport.TopicSaveBlock:             dp.saveBlock,
+		outport.TopicRevertIndexedBlock:    noOpHandler,
```

Comment:
maybe you don't need here to add the list with all the topics. 
In case of an unknown topic from the map, you can skip the processing of the payload.

Reply:
yup, can be handled in the ProcessPayload function

Reply:
Right, very good catch here.
Fixed it as suggested